### PR TITLE
Idir instead of GitHub

### PR DESCRIPTION
**Pull request by ShovelHand**
This change involves getting IDIR info for contacts on project requests instead of GitHub.
image
 
The best way to test is to start writing in the last name of a IDIR user in the appropriate box, and we should get back their user info and a user Photo if they have on. Looks like most users don't have one. Try typing in 'mitovska' to see our wonderful boss' face an example of this working properly. 

There are a few user experience issues I've created separate tickets to address. These are:
- having a better placeholder for missing user photos instead of a broken image link and alt text. 
- Let's get rid of the "Idir Id" field altogether, and just use the email address to get the user. I haven't done this yet because it involves a bit of back-end work, and it's time to move this task along.

### CODE REVIEW

**Comment by ShovelHand** [comment]
@ty2k, @okanji Would you guys mind having a look? Tyler is most familiar with the changes made since he gave me so much help, but it's a big PR, and another pair of eyes is a good idea.

**Comment by ty2k** [comment]
I haven't been able to test this in the lab today due what looks like some kind of infinite loop of network requests in the new product creation flow. I'm not sure if this is a connection issue on my end, or some circular logic in the React component (ex: a network request in the component is setting state in a way that is ultimately changing the component's own props, causing it to render over and over). I think this might be what you were talking to Oamar about, but I'm not 100% sure:

Screen Shot 2022-11-17 at 4 22 09 PM

I notice that I don't experience this issue in the product edit flow. However, it looks like we need to update this view to account for the new IDIR requirements (dropping references to GitHub if users are unable to select GitHub users, etc).

Screen Shot 2022-11-17 at 3 50 29 PM

File: web/src/redux/githubID/graphAuthConfig.js
Comment (ty2k) [inline web/src/redux/githubID/graphAuthConfig.js:8]: Let's access these values from environment variables set in `.env` instead. Since this React application was scaffolded with Create React App, we need to follow their way of doing this: https://create-react-app.dev/docs/adding-custom-environment-variables/

Your environment variables in `.env` should look like `REACT_APP_AZURE_CLIENT_ID` and `REACT_APP_AZURE_TENANT_ID`. The `REACT_APP_` prefix is necessary for any environment variables to show up in an app created with Create React App.

File: .env.example
Comment (ty2k) [inline .env.example:33]: I'm see this addition to the `.env.example`, but I don't see these variables being used, except when they're passed through the `docker-compose` file. I think it's a good idea to leave these here and use them in the React front-end.

**Comment by ty2k** [comment]
Can you please update the placeholder text in `<GitHubUserValidation>` to indicate that we need an IDIR email address as opposed to a GitHub username?

Screenshot 2022-11-23 at 2 39 39 PM

**Comment by ty2k** [comment]
I've added an example branch [graph-api-response-logic](https://github.com/bcgov/platform-services-registry/commits/graph-api-response-logic) with two commits where I updated `searchIdirUsers()` in `githubID.action.ts` and the `GITHUBID_USER_DOES_NOT_EXIST` case in `githubID.reducer.ts` to bring back the "No user found :(" message when a search fails, as opposed to the current state with leaves the "Loading..." message there.

I'd recommend doing something like this to let users know that the search was completed but no results were found. If we were to deploy with the "Loading..." message staying up after a search finishes, it will cause confusion.

ProfileCreate when searching for a user that doesn't exist

**Comment by ty2k** [comment]
There are at least three leftover labels referencing GitHub usernames that we need to update:

- [ ] Update the "Write a github username" placeholder text in `GithubUserValidation.tsx`
- [ ] Replace references to "GitHub Id" in `ContactCardEdit.tsx`
- [ ] Replace reference to "Github IDs" in `PublicLanding.tsx`

File: web/package.json
Comment (ty2k) [inline web/package.json:33]: We're importing the minified distribution code from the `bootstrap` package in `web/src/index.tsx`, but we don't actually import `react-bootstrap` anywhere. `react-bootstrap` should be left out unless we're using the components from that library directly, I think.

File: web/src/views/ProfileCreate.tsx
Comment (ty2k) [inline web/src/views/ProfileCreate.tsx:169]: I'm assuming you won't be using this commented code for anything, right? It should be left out unless it's necessary.

**Review by ty2k — CHANGES_REQUESTED** [review]
Thanks for the hard work on this major functional change, @ShovelHand!

I have tested the Registry stack locally in its current state using these manual test cases:

1. Try to submit a new request with known good IDIRs for the PO, TL1, and TL2. (Should succeed)
2. Try to submit a new request with known good IDIRs for the PO and TL1. (Should succeed)
3. Try to submit a new request with known good IDIRs for the PO and TL1, but not for TL2. (Should not be possible)
4. Try to submit a new request with known good IDIR for the PO, but not for TL1. (Should not be possible)
5. Try to submit an edit request with known good IDIRs for the PO, TL1, and TL2. (Should succeed)
6. Try to submit an edit request with known good IDIRs for the PO and TL1. (Should succeed)
7. Try to submit an edit request with known good IDIR for the PO and TL1, but not for TL2. (Should not be possible)
8. Try to submit an edit request with known good IDIR for the PO, but not for TL1. (Should not be possible)

I find that all of these manual test cases work with the code at this point.

That being said, we should address the issues I listed in the thread above before deploying. We should also finish [#3099 Document Rollout Strategy for replacing Github IDs for IDIR for POs and TLs in Registry](https://app.zenhub.com/workspaces/platform-experience-5bb7c5ab4b5806bc2beb9d15/issues/bcdevops/developer-experience/3099) ahead of deploying this, I assume?

It might also be good to deploy [#3147 Gracefully handle missing photos when Idir user added to Project Request in Platform Services Registry](https://app.zenhub.com/workspaces/platform-experience-5bb7c5ab4b5806bc2beb9d15/issues/bcdevops/developer-experience/3147) before or at the same time as this change, since the large majority of IDIR users aren't going to have avatars (vs GitHub where every user gets an auto-generated pixel art avatar). This should be a straightforward change compared to the much more involved work in this PR. 😁

File: web/src/__tests__/ProfileCreate.test.tsx
Comment (w8896699) [inline web/src/__tests__/ProfileCreate.test.tsx:24]: Please try not to merge in comment out code, and also it seems like you disable this test, you can probably disable/skip this test instead. 
Last, please squash before merge to master branch

**Review by w8896699 — CHANGES_REQUESTED** [review]

**Pull request merged**
